src/app/route/MyAccountWishlist/MyAccountWishlist.component.js

```diff
@@ -0,0 +1,176 @@
+            removeProductFromWishlist
+        } = this.props;
+        this.setState({ isLoading: true });
+
+
+        if (!isSignedIn()) {
+            showNotification('error', 'You must login or register to add all products to Cart.');
+            this.setState({ isLoading: false });
+            return null;
+        }
+
+        const data = [];
+        data.errors = [];
+        data.addedProducts = [];
+        return Promise.all(Object.values(wishlistItems).map((product) => {
+            const { type_id, configurableVariantIndex, name } = product;
+            if (type_id === 'configurable' && !configurableVariantIndex) {
+                data.errors.push(`You need to choose options for your item for ${name}`);
+                return null;
+            }
+
+            return addProduct({
+                product,
+                quantity: 1
+            }).then(() => data.addedProducts.push(name) && removeProductFromWishlist({ product, noMessages: true }));
+        })).then(() => this.afterAdded(data));
+    }
+
+    renderWishlistItem(product, isUpdatingWishlist) {
+        if (isUpdatingWishlist) {
+            return (
+                <ProductCard product={ {} } key={ product.id } />
+            );
+        }
+
+        return (
+            <ProductCard
+              product={ product }
+              key={ product.id }
+              wishlistItem
+            />
+        );
+    }
+
+    render() {
+        const {
+            wishlistItems,
+            isUpdatingWishlist
+        } = this.props;
+
+        const { isLoading } = this.state;
+
+        if (!isSignedIn()) {
+            return <Redirect to="/" />;
+        }
+
+        return (
+            <main block="MyAccountWishlist" aria-label="My Account Wishlist">
+                <div block="MyAccountWishlist" elem="Wrapper">
+                    <MyAccountSidebar />
+                    <div block="MyAccountWishlist" elem="Content">
+                        <h2>My Wish List</h2>
+                        <Loader isLoading={ isLoading } />
+                        {
+                            Object.keys(wishlistItems).length > 0
+                                ? (
+                                    <>
+                                        <button
+                                          onClick={ () => this.addToCart() }
+                                          block="MyAccountWishlist"
+                                          elem="Button"
+                                          mods={ { isLoading } }
+                                          disabled={ isLoading }
+                                        >
+                                            Add All To Cart
+                                        </button>
+                                        <ul block="MyAccountWishlist" elem="List">
+                                            {
+                                                Object.values(wishlistItems).map(
```

Comment:
Let's move this to previous line, `{  Object.values(wishlistItems).map(` might be more readable, I don't like almost blank lines in JSX template.